src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -0,0 +1,178 @@
+            + "</code></pre>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_multipleComments() {
+    // Add methods, like `addComment()`, should add components at any place,
+    // and they will get printed in order.
+    String comment1 = "This is a test comment.";
+    String comment2 = "This is unordered list.";
+    String comment3 = "This is ordered list.";
+    List<String> orderedList =
+        Arrays.asList("A flattened method.", "A request object method.", "A callable method.");
+    List<String> unOrderedList =
+        Arrays.asList("A flattened method.", "A request object method.", "A callable method.");
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder()
+            .addComment(comment1)
+            .addComment(comment2)
+            .addUnorderedList(unOrderedList)
+            .addComment(comment3)
+            .addOrderedList(orderedList)
+            .build();
+    String expected =
+        "This is a test comment.\n"
+            + "This is unordered list.\n"
+            + "<ul>\n"
+            + "<li> A flattened method.\n"
+            + "<li> A request object method.\n"
+            + "<li> A callable method.\n"
+            + "</ul>\n"
+            + "This is ordered list.\n"
+            + "<ol>\n"
+            + "<li> A flattened method.\n"
+            + "<li> A request object method.\n"
+            + "<li> A callable method.\n"
+            + "</ol>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_multipleParams() {
+    // Parameters should be grouped together and get printed after block comments.
+    String comment = "This is a block comment.";
+    String paramName1 = "shelfName";
+    String paramDescription1 = "The name of the shelf where books are published to.";
+    String paramName2 = "shelfId";
+    String paramDescription2 = "The shelfId of the shelf where books are published to.";
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder()
+            .addParam(paramName1, paramDescription1)
+            .addParam(paramName2, paramDescription2)
+            .addComment(comment)
+            .build();
+    String expected =
+        "This is a block comment.\n"
+            + "@param shelfName The name of the shelf where books are published to.\n"
+            + "@param shelfId The shelfId of the shelf where books are published to.";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_ThrowsDeprecated() {
+    // No matter how many times or order `setThrows` and `setDeprecated` are called,
+    // only one @throws and @deprecated will be printed.
+    String throwType = "com.google.api.gax.rpc.ApiException";
+    String throwsDescription = "if the remote call fails.";
+    String throwType_print = "java.lang.RuntimeException";
+    String throwsDescription_print = "if the remote call fails.";
```

Comment:
Nit: Java names generally use lowerCamelCase. The `methodTest_edgeCase()` convention is an exception only for test methods.

Reply:
rechecked the test methods name, thanks for catching this!